azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_provider.py

```
@@ -80,10 +80,19 @@ def on_subscribe_callback(client, userdata, mid):
                 logger.error("Unexpected error calling on_mqtt_subscribed")
                 logger.error(traceback.format_exc())
 
+        def on_message_callback(client, userdata, mqtt_message):
+            logger.info("message received")
+            try:
+                self.on_mqtt_message_received(mqtt_message._topic, mqtt_message.payload)
+            except:  # noqa: E722 do not use bare 'except'
+                logger.error("Unexpected error calling on_mqtt_message_received")
+                logger.error(traceback.format_exc())
```

should we re-raise the exception here after logging it?